lms/djangoapps/learner_dashboard/urls.py

```diff
@@ -6,5 +6,6 @@
 from . import views
 
 urlpatterns = [
+    url(r'^programs/(?P<programId>[0-9]+)/$', views.view_program_details, name='program_details_view'),
```

Comment:
@AlasdairSwan a UUID can consist of the digits 0-9, letters a-f, and hyphens. `[0-9a-z-]+` should get the job done. Here's one way to verify:
```python
>>> import re
>>> import uuid
>>>
>>> id = str(uuid.uuid4())
>>> id
'd7e176a5-a2c3-4865-847c-bbca06243f78'
>>> pattern = re.compile(r'[0-9a-z-]+')
>>> r = pattern.match(id)
>>> r.group(0)
'd7e176a5-a2c3-4865-847c-bbca06243f78'
>>> 
```
Note that if we want to key off of UUID here, we're going to need to make some changes to the programs API. It doesn't currently allow program lookup by UUID, nor does it include the UUID in any data it returns.